### PR TITLE
cn: introduce PeerSet interface to test ProtocolManager

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- Introduce `PeerSet` interface to test ProtocolManager
- To test ProtocolManager, `peerSet` should be mocked, hence `PeerSet` interface is introduced.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ehnuje** [comment]
With this change and the upcoming mocking, ProtocolManager test does not have to consider peerSet while its tests. `peerSet` will be tested by its own unit tests.

File: node/cn/handler.go [hunk 1]
```diff
@@ -258,8 +258,8 @@ func NewProtocolManager(config *params.ChainConfig, mode downloader.SyncMode, ne
 }
 
 // istanbul BFT
```
Comment (jeonghonah) [inline node/cn/handler.go]: What is the meaning of this comment?

File: node/cn/peer_set.go [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+	delete(ps.peers, id)
+	p.Close()
+
+	cnPeerCountGauge.Update(int64(len(ps.cnpeers)))
+	pnPeerCountGauge.Update(int64(len(ps.pnpeers)))
+	enPeerCountGauge.Update(int64(len(ps.enpeers)))
+	return nil
+}
+
+// istanbul BFT
```
Comment (jeonghonah) [inline node/cn/peer_set.go]: (duplicated) Do I know the meaning of this comment?

File: node/cn/handler.go (hunk 1 above)
Comment (ehnuje) [reply node/cn/handler.go]: I think that this means this is not something generally used method, but the method for Istanbul BFT. If we don't use Istanbul BFT, we don't need to manage the list of validators.

File: node/cn/peer_set.go (hunk 2 above)
Comment (ehnuje) [reply node/cn/peer_set.go]: This one can be removed.

File: node/cn/peer_set.go [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+	for _, p := range ps.peers {
+		if _, td := p.Head(); bestPeer == nil || td.Cmp(bestTd) > 0 {
+			bestPeer, bestTd = p, td
+		}
+	}
+	return bestPeer
+}
+
+// RegisterValidator registers a validator.
+func (ps *peerSet) RegisterValidator(connType p2p.ConnType, validator p2p.PeerTypeValidator) {
```
Comment (jeonghonah) [inline node/cn/peer_set.go]: Can I know a reason to introduce `RegisterValidator` to the peerSet?

Comment (ehnuje) [reply node/cn/peer_set.go]: @jeonghonah Sure. Before the change, `ProtocolManager` can directly access to `ChannelManager` of `peerSet`, however, as `peerSet` has been changed into `PeerSet` interface, we need an additional interface, `RegisterValidator`.

**Pull request merged**